modules/smtp/modules.php

```diff
@@ -1528,6 +1528,7 @@ function get_primary_recipient($profiles, $headers, $smtp_servers, $is_draft=Fal
 if (!hm_exists('delete_draft')) {
 function delete_draft($id, $cache, $imap_server_id, $folder) {
     $imap = Hm_IMAP_List::connect($imap_server_id);
+    if ( ! $imap ) return false;
```

Comment:
This is a result of incorrectly setup imap server which is used in a profile but cannot be authenticated (wrong password maybe). At any rate, I agree we should check if imap is authenticated before trying to select a mailbox. Can you use this, instead:
```
if (! imap_authed($imap)) {
    return false;
}
```